EpsilonAuxBMS/Src/main.c

```diff
@@ -512,10 +506,15 @@ void HAL_CAN_RxCpltCallback(CAN_HandleTypeDef* hcan)
 
     CanRxMsgTypeDef* msg = hcan->pRxMsg;
 
-    if (msg->StdId == ORION_MAX_MIN_VOLTAGES_STDID)
+    if (msg->StdId == ORION_MAX_MIN_VOLTAGES_STDID && msg->DLC == 8)
     {
-        orionStatus.maxCellVoltage = (uint16_t)msg->Data[6]; // Max Cell voltage
-        orionStatus.minCellVoltage = (uint16_t)msg->Data[4]; // Min Cell Voltage
+        // Voltages are 2 bytes each, and memory is stored in little endian format
+        orionStatus.maxCellVoltage = (uint16_t)msg->Data[6] << 8 | msg->Data[7]; // Max Cell voltage
```

Comment:
Messages are in little endian format, but they're being received in little endian format as well. So it shoould be a straight conversion. (Data[0] & Data[2])